.github/workflows/continuous-delivery.yml

```diff
@@ -1,84 +1,96 @@
-      SSH_PRIVATE_KEY: '${{ secrets.SSH_PK }}'
-      SSH_USER: '${{ secrets.SSH_USER }}'
+name: Continuous Delivery
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [testing, production]
+
+permissions:
+  contents: read
+  packages: write
+
+env:
+  # Uses docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+  REPOSITORY_NAME: ${{ github.event.repository.name }}
+
+jobs:
+  Publish:
+    name: Publish image to GHCR 🚀
+    runs-on: ubuntu-latest
+    outputs:
+      deploy_tag: ${{ steps.save_tag.outputs.tag }}
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          submodules: true
+
+      - name: Extract version from package.json
+        id: package_version
+        uses: saionaro/extract-package-version@v1.2.1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2.1.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata for Docker images
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=raw,value=${{ steps.package_version.outputs.version }}
+            type=raw,value=${{ steps.package_version.outputs.version }}.${{ github.run_number }}
+            type=raw,value=latest,enable=${{ endsWith(github.ref, github.event.repository.default_branch) }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4.0.0
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Save image tag for use in deploy job
+        id: save_tag
+        run: echo "::set-output name=tag::${{ steps.package_version.outputs.version }}.${{ github.run_number }}"
+  Deploy_Prod:
+    name: Deploy Production Bots 🚀
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/production'
+    needs: Publish
+    steps:
+      - name: Bithdayy 🎂
+        uses: ./.github/workflows/deploy.yml
+        env:
+          DOCKER_TAG: 'production'
+          DOPPLER_TOKEN: ${{ secrets.DOPPLER_TOKEN_PRODUCTION }}
+      - name: StrayKids 🧒
+        uses: ./.github/workflows/deploy.yml
+        env:
+          DOCKER_TAG: 'stray-kids'
+          DOPPLER_TOKEN: ${{ secrets.DOPPLER_TOKEN_STRAYKIDS }}
+      - name: Birthdayy Custom 🤡
+        uses: ./.github/workflows/deploy.yml
+        env:
+          DOCKER_TAG: 'custom-bot'
+          DOPPLER_TOKEN: ${{ secrets.DOPPLER_TOKEN_BIRTHDAYY_CUSTOM }}
+  Deploy_Test:
+    name: Deploy Testing Bot 🧪
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/testing'

```

Comment:
die his makes it impossible to run this workflow somewhere else than testing and I think is this a problem.

Reply:
This is exactly what we are currently doing...

Reply:
it currently is, just run the file from the dev branch an the dev version will be used for the deployment